bracex/__init__.py

```diff
@@ -38,46 +44,46 @@ class ExpansionLimitException(Exception):
     """Brace expansion limit exception."""
 
 
-def expand(string, keep_escapes=False, limit=DEFAULT_LIMIT):
+def expand(string: String, keep_escapes: bool = False, limit: int = DEFAULT_LIMIT) -> List[String]:
     """Expand braces."""
 
     return list(iexpand(string, keep_escapes, limit))
 
 
-def iexpand(string, keep_escapes=False, limit=DEFAULT_LIMIT):
+def iexpand(string: String, keep_escapes: bool = False, limit: int = DEFAULT_LIMIT) -> Iterator[String]:
     """Expand braces and return an iterator."""
 
     if isinstance(string, bytes):
         is_bytes = True
-        string = string.decode('latin-1')
-
+        value = string.decode('latin-1')
     else:
+        value = string
         is_bytes = False
 
-    for entry in ExpandBrace(keep_escapes, limit).expand(string):
-        yield entry.encode('latin-1') if is_bytes else entry
+    for entry in ExpandBrace(keep_escapes, limit).expand(value):
+        yield entry.encode('latin-1') if is_bytes else entry  # type: ignore
```

Comment:
@henryiii One last question. I decided to go ahead and type the whole library as a learning experience (as the library is a fairly small and simple library), and I have it all working, but this line I had to ignore. Is there a good way to handle this aside from ignoring it?

Basically, we take either bytes or str and the return will be an Iterator of that same type. ExpandBrace takes a string, so we convert it beforehand and then have to convert return from ExpandBrace back to the type. Mypy doesn't seem to be "smart" enough to see that `is_bytes`'s value was based on our earlier type check, so it gets confused. 

If I do the following, it passes, but that is a waste of time to do. Is ignoring really the best way?

```py
    for entry in ExpandBrace(keep_escapes, limit).expand(value):
        if isinstance(string, bytes):
            yield entry.encode('latin-1')
        else:
            yield entry
```